src/api/app/views/webui2/webui/projects/pulse/_pulse_list.html.haml

```diff
@@ -12,81 +12,82 @@
       = render partial: 'pulse_list_comments', locals: { comments: comments }
 
   .col
-    .card
-      .card-header
-        Requests
-      .card-body
-        - if requests.any?
-          .row
-            .col
-              .progress
-                -# haml-lint:disable InlineStyles
-                - requests_by_percentage.each do |state, percentage|
-                  %div{ title: "#{requests_by_state[state]} #{state} requests",
-                        class: "progress-bar progress-state-#{state}",
-                        'aria-valuemax': '100', 'aria-valuemin': '0', 'aria-valuenow': percentage,
-                        role: 'progressbar', style: "width: #{percentage}%" }
-                -# haml-lint:enable InlineStyles
-              %p
-                = link_to('#pulse-requests') do
-                  = requests_by_state.values.sum
-                  active requests
-          .row
-            - requests_by_state.each_key do |state|
-              .col.border.text-center
-                %p.fa-3x
-                  %i{ class: "fa #{request_bootstrap_icon(state)} request-state-#{state}" }
-                %p
-                  = pluralize(requests_by_state[state], 'request')
-                  %br
-                  in #{state}
-        - else
-          = link_to(project_requests_path(project)) do
-            There have been no requests send to this project.
+    = render partial: 'pulse_list_requests_box', locals: { project: project,
+                                                           requests: requests,
+                                                           requests_by_percentage: requests_by_percentage,
+                                                           requests_by_state: requests_by_state }
+
 .row
   .col
-    %h5#pulse-packages
-      Package Changes
-    %hr
-    %ul.list-unstyled
-      - (commits + new_packages + deleted_packages + updates).sort_by(&:datetime).reverse_each do |log_entry|
-        %li
-          = render partial: 'pulse_list_entry', locals: { log_entry: log_entry, project: project }
+    - package_changes = commits + new_packages + deleted_packages + updates
+    - if package_changes.present?
+      %h5#pulse-packages
+        Package Changes
+        %small
+          = link_to('.collapse-package-changes', aria: { controls: 'collapse-package-changes', expanded: 'true' },
+                    'data-toggle' => 'collapse', role: 'button') do
```

Comment:
I think you're missing `data-target` which makes all but the last collapsed. I would also link the whole h5, not only the hide/show words.

Reply:
`data-target` is needed only if `href` is not defined. [See the documentation from Bootstrap](https://getbootstrap.com/docs/4.1/components/collapse/#example):

> You can use a link with the href attribute, or a button with the data-target attribute. In both cases, the data-toggle="collapse" is required.

Did you mean that it would be nice to collapse/expand all of the `div` at once? Just making sure I understand you correctly.

For the link on the whole `h5`, I'm fine with that.

Reply:
Oh, I forgot the `show` class